From d0ff7ef0f932d99dd0b76a21aab4f40eedce9413 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Wed, 10 Mar 2021 06:48:18 -0800
Subject: [PATCH] Revert "storage: Remove unused
 __connman_storage_open_service()"

This reverts commit 6ca9222beb5445de9fc8a94d92e0a7b9ebce3724.

"Add p2p changes" patch from LGE is still using this
---
 src/connman.h |  1 +
 src/storage.c | 22 ++++++++++++++++++++++
 2 files changed, 23 insertions(+)

diff --git a/src/connman.h b/src/connman.h
index dac252d9..e23dbc13 100644
--- a/src/connman.h
+++ b/src/connman.h
@@ -272,6 +272,7 @@ void __connman_storage_delete_global(void);
 GKeyFile *__connman_storage_load_config(const char *ident);
 GKeyFile *__connman_storage_load_provider_config(const char *ident);
 
+GKeyFile *__connman_storage_open_service(const char *ident);
 int __connman_storage_save_service(GKeyFile *keyfile, const char *ident);
 GKeyFile *__connman_storage_load_provider(const char *identifier);
 void __connman_storage_save_provider(GKeyFile *keyfile, const char *identifier);
diff --git a/src/storage.c b/src/storage.c
index 7e6d6590..2de5c7f8 100644
--- a/src/storage.c
+++ b/src/storage.c
@@ -161,6 +161,28 @@ GKeyFile *__connman_storage_load_provider_config(const char *ident)
 	return keyfile;
 }
 
+GKeyFile *__connman_storage_open_service(const char *service_id)
+{
+	gchar *pathname;
+	GKeyFile *keyfile = NULL;
+
+	pathname = g_strdup_printf("%s/%s/%s", STORAGEDIR, service_id, SETTINGS);
+	if (!pathname)
+		return NULL;
+
+	keyfile =  storage_load(pathname);
+	if (keyfile) {
+		g_free(pathname);
+		return keyfile;
+	}
+
+	g_free(pathname);
+
+	keyfile = g_key_file_new();
+
+	return keyfile;
+}
+
 gchar **connman_storage_get_services(void)
 {
 	struct dirent *d;
